File: level1/lib/detection/person/mock_person_engine.dart
import '../../core/models.dart';
import 'person_engine.dart';

/// Test/dry-run engine: returns whatever [persons] was pre-loaded with.
class MockPersonEngine implements PersonEngine {
  final List<PersonBox> persons = [];

  @override
  Future<void> init() async {}

  @override
  Future<List<PersonBox>> detectPersons(ColorBitmap frame) async =>
      List.of(persons);

  @override
  Future<void> dispose() async {}
}

File: android/app/src/test/kotlin/io/securitycam/level2/detection/person/MockPersonEngine.kt
package io.securitycam.level2.detection.person

import io.securitycam.level2.detection.DetectedBox

/** Test/dry-run engine: returns whatever [persons] was pre-loaded with. */
class MockPersonEngine : PersonEngine {
    val persons = mutableListOf<DetectedBox>()

    override suspend fun init() {}

    override suspend fun detectPersons(frame: io.securitycam.level2.detection.ColorBitmap): List<DetectedBox> =
        persons.toList()

    override suspend fun dispose() {}
}
